EventSource/src/EventSourceUtil.ts

```diff
@@ -57,6 +57,129 @@ export function getTimeStamp(resource: Resource, timestampPath: string): number
     return extractTimestampFromLiteral(resourceStore.getObjects(null, timestampPath, null)[0] as Literal)// Note: expecting real xsd:dateTime
 }
 
+export async function prefixesFromFilepath(path: string, url?: string): Promise<any> {
+    let prefixes = {};
+    if (url) {
+        prefixes[""] = url + "#";
+    }
+    if (existsSync(path)) {
+        const store = await turtleStringToStore(readFileSync(path, "utf-8"));
+        // only the triples using predicate "<http://purl.org/vocab/vann/preferredNamespacePrefix>"
+        // are relevant, as these represent prefix (= object) and URI (= subject)
+        const prefixQuads = store.getQuads(null, namedNode("http://purl.org/vocab/vann/preferredNamespacePrefix"), null, null);
+        for (const prefixQuad of prefixQuads) {
+            if (prefixQuad.object.termType != "Literal" || ! /^"[^"]+"$/.test(prefixQuad.object.id)) {
+                // the object does not represent a string literal, skipping this entry
+                continue;
+            }
+            prefixes[prefixQuad.object.id.substring(1, prefixQuad.object.id.length - 1)] = prefixQuad.subject.id;
+        }
+    }
+    return prefixes;
+}
+
+type ResourceMap = Map<string, Map<string, Quad_Object[]>>;
+
+/**
+ * Converts a resource (quad array) to two maps, where every key is a subject ID.
+ * The maps are split depending on wether the subject node is a named node or a
+ * blank node
+ * Note: currently duplicate objects are not checked. As most functions
+ *  use this method to optimise the utilized space, it might be beneficial to remove
+ *  duplicate objects here as well
+ * @param resource
+ * @returns {[ResourceMap, ResourceMap]}
+ */
+export function resourceToMaps(resource: Resource): [ResourceMap, ResourceMap] {
```

Comment:
Could you explain me why this method is necessary. I don't quite understand the need of it.

Reply:
This has now been changed to be inlined instead, removing the separate extra function.